lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -859,6 +860,9 @@ def _create_deployment_parameters(
             if azure_node_runbook.vhd:
                 # vhd is higher priority
                 azure_node_runbook.marketplace = None
+                azure_node_runbook.shared_gallery = None
+            elif azure_node_runbook.shared_gallery:
+                azure_node_runbook.marketplace = None
             elif not azure_node_runbook.marketplace:
```

Comment:
Change it to else may be more clear.

Reply:
Sorry I didn't get this. Do you mean move ` elif azure_node_runbook.shared_gallery` to an `else`?

Reply:
yes, it would be more clear.

Reply:
How about this comment?

Reply:
I had updated the code as per this conversation. Can you please take a look and confirm?

Reply:
Sorry, I misread the logic here. The logic here should be like below, and it needs a comments for else block. So the logic looks like not missing anything

```python
elif azure_node_runbook.shared_gallery:
    azure_node_runbook.marketplace = None
elif azure_node_runbook.marketplace:
    # take the value of azure_node_runbook.marketplace 
    ...
else:
    # set to default marketplace, if nothing specified
    azure_node_runbook.marketplace = AzureVmMarketplaceSchema()
```

Reply:
This logic will not work because `azure_node_runbook.marketplace` might have some preexisting value and this initialization will overwrite. We want to initialize with default value if `azure_node_runbook.marketplace` is empty. So the whole logic will look something like : 

```
elif azure_node_runbook.shared_gallery:
    azure_node_runbook.marketplace = None
elif azure_node_runbook.marketplace:
    # take the value of azure_node_runbook.marketplace 
    ...
else:
    # set to default marketplace, if nothing specified
    if not azure_node_runbook.marketplace:
           azure_node_runbook.marketplace = AzureVmMarketplaceSchema()
```

Reply:
The `elif azure_node_runbook.marketplace` will be empty. we don't need to do anything if `azure_node_runbook.marketplace` is already filled. I can add a `pass` there to make the overall logic more clear.

Reply:
`...` is the same as `pass`. yes, it's empty. Sometime people are wondering if there miss an `else`. The comments help to explain the empty clause, and make the logic more straightforward (use elif for values, and else for all None).

Reply:
Definitely! The changes you suggested make the logic way more clear. Thanks for this suggestion!